torch/csrc/jit/ir/ir.cpp

```diff
@@ -1608,17 +1608,25 @@ Node* Graph::createTupleIndex(
   return n;
 }
 
-Node* Graph::createTupleSlice(Value* tup, int64_t beg, int64_t end) {
-  auto n = create(prim::TupleSlice, {tup});
-  auto tuple_type = tup->type()->expect<TupleType>();
-  n->i_(attr::beg, beg);
-  n->i_(attr::end, end);
-  std::vector<TypePtr> output_types;
-  for (auto i = beg; i < end; ++i) {
-    output_types.push_back(tuple_type->elements().at(i));
-  }
-  auto tt = TupleType::create(std::move(output_types));
-  n->output()->setType(tt);
+Node* Graph::createTupleSlice(
+    Value* tup,
+    int64_t beg,
+    int64_t end,
+    int64_t step_size) {
+  auto unpacked_tuple = insertNode(createTupleUnpack(tup));
```

Comment:
I don't think you need to use `TupleUnpack` here, which reads out all values from tuple. Instead, I think `prim::TupleIndex` is better.